homeassistant/components/heos/config_flow.py

```diff
@@ -23,8 +23,13 @@ class HeosFlowHandler(config_entries.ConfigFlow):
 
     async def async_step_discovery(self, discovery_info):
         """Handle a discovered Heos device."""
-        return await self.async_step_user(
-            {CONF_HOST: discovery_info[CONF_HOST]})
+        # Only continue if this is the only active flow
+        flows = self.hass.config_entries.flow.async_progress()
+        heos_flows = [flow for flow in flows if flow['handler'] == DOMAIN]
+        if len(heos_flows) == 1:
+            return await self.async_step_user(
+                {CONF_HOST: discovery_info[CONF_HOST]})
+        return self.async_abort(reason='already_setup')
```

Comment:
Having only used config flow stuff sparingly and very recently, I feel like this reason has to be caught somewhere to be able to show the user the proper message in the frontend... or is this a built in one?

Reply:
It won't show anything, but there's no need to show anything.  Connecting to 1 HEOS devices enables controlling all of them (hence why we only need a single config entry in the end).  Discovery detects all of the devices individually, and once we've setup on one, we don't care about the rest of the discovered items.